File: tests/xcalablemp/global-view/gmove/Fortran/gmove_G2L_1a2t_bc.f90
program tgmove
integer,parameter :: n=4
integer a(n,n), b(n,n)
integer xmp_node_num
!$xmp nodes p(2,2)
!$xmp template tx(n,n)
!$xmp distribute tx(cyclic(2),cyclic(2)) onto p
!$xmp align a(*,i1) with tx(*,i1)

!$xmp loop (i1) on tx(*,i1)
do i1=1,n
  do i0=1,n
    a(i0,i1)=i0+i1
  end do
end do

do i1=1,n
  do i0=1,n
    b(i0,i1)=0
  end do
end do

!$xmp gmove
b(2:n,2:n)=a(2:n,2:n)

ierr=0
do i1=2,n
  do i0=2,n
    ierr=ierr+abs(b(i0,i1)-i0-i1)
  end do
end do

!$xmp reduction (max:ierr)
irank=xmp_node_num()
if (irank==1) then
  print *, 'max error=',ierr
endif
!call exit(ierr)

stop
end program tgmove

File: tests/xcalablemp/global-view/gmove/C/gmove_G2L_1a2t_bc.c
#define N 2
#include <stdio.h>
#include <math.h>

extern int chk_int(int ierr);

int n=N;
double a[n][n];
#pragma xmp nodes p(2,2)
#pragma xmp template tx(0:n-1,0:n-1)
#pragma xmp distribute tx(cyclic(2),cyclic(2)) onto p
#pragma xmp align a[*][i] with tx(*,i)

int main(){

  int i0,i1,ierr;
  double b[N][N],err;

  for(i0=0;i0<n;i0++){
#pragma xmp loop (i1) on tx(*,i1)
    for(i1=0;i1<n;i1++){
      a[i0][i1]=i0+i1+1;
    }
  }

  for(i0=0;i0<n;i0++){
    for(i1=0;i1<n;i1++){
      b[i0][i1]=0.0;
    }
  }

#pragma xmp gmove
  b[1:n-1][1:n-1]=a[1:n-1][1:n-1];

  err=0.0;
  for(i0=1;i0<n;i0++){
    for(i1=1;i1<n;i1++){
      err=err+fabs(b[i0][i1]-i0-i1-1);
    }
  }

#pragma xmp reduction (MAX:err)
  ierr=err;
  chk_int(ierr);
}
